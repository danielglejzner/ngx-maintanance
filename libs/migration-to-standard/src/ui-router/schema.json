{
  "$schema": "https://json-schema.org/schema",
  "$id": "",
  "title": "",
  "type": "object",
  "properties": {
  },
  "required": []
}
