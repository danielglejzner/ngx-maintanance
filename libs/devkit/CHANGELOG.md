## 0.1.1 (2024-04-06)


### 🩹 Fixes

- update documentation ([d2968b2](https://github.com/danielglejzner/ngx-maintenance/commit/d2968b2))


### ❤️  Thank You

- Arturo Volpe @aVolpe
- Jordan Hall @Jordan-Hall

## 0.1.0 (2024-03-13)


### 🚀 Features

- import angular material rail drawer ([1f81f54](https://github.com/danielglejzner/ngx-maintenance/commit/1f81f54))

- use cli to get from package json or github URL ([f9d4ea4](https://github.com/danielglejzner/ngx-maintenance/commit/f9d4ea4))

- **npm-api:** refactor npm registry api into a standalone lib and strongly typed ([cc10093](https://github.com/danielglejzner/ngx-maintenance/commit/cc10093))


### ❤️  Thank You

- Jordan Hall @Jordan-Hall