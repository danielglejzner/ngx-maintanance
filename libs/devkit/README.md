# devkit

Our devkit is designed to help develop and support the Angular Compatibility Maintenance Initiative efforts