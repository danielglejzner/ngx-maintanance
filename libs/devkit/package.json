{
  "name": "@ngx-maintenance/devkit",
  "version": "0.3.0",
  "dependencies": {
    "@ngx-maintenance/utils": "latest",
    "@ngx-maintenance/npm-api": "latest",
    "tslib": "^2.3.0"
  },
  "type": "commonjs",
  "main": "./src/index.js",
  "typings": "./src/index.d.ts"
}
