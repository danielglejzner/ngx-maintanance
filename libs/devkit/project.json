{
  "name": "devkit",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/devkit/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/devkit",
        "main": "libs/devkit/src/index.ts",
        "tsConfig": "libs/devkit/tsconfig.lib.json",
        "assets": ["libs/devkit/*.md"]
      }
    },
    "publish": {
      "command": "node tools/scripts/publish.mjs devkit {args.ver} {args.tag}",
      "dependsOn": ["build"]
    }
  },
  "tags": []
}
