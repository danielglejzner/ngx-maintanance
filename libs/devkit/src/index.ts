export * from './lib/to-maintenance-initiative';
export { checkoutImportAndMigrateAngular, } from './lib/checkout-and-update'
export { httpsGet, httpsRequest, tryCatch } from '@ngx-maintenance/utils'