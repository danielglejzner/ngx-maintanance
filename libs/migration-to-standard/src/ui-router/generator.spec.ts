import { Tree, addProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

import migrateToAngularRouter from './generator';
import { GeneratorSchema } from './schema';

// Thank you to ngxtension for helping with setup https://github.com/nartc/ngxtension-platform/blob/main/libs/plugin/src/generators/convert-signal-inputs/generator.spec.ts
const template = `
  <div #dialogdiv></div>
    <div class="navheader">
      <ul *ngIf="isAuthenticated" class="nav nav-tabs">

        <li uiSrefActive="active"> <a uiSref="mymessages" role="button"> Messages </a> </li>
        <li uiSrefActive="active"> <a uiSref="contacts" role="button"> Contacts </a> </li>
        <li uiSrefActive="active"> <a uiSref="prefs" role="button"> Preferences </a> </li>

        <li class="navbar-right">
          <button class="btn btn-primary fa fa-home" uiSref="home"></button>
          <button style="margin-right: 15px;" class="btn btn-primary" uiSref="mymessages.compose">
            <i class="fa fa-envelope"></i> New Message
          </button>
        </li>

        <li class="navbar-text navbar-right logged-in-user" style="margin: 0.5em 1.5em;">
          <div>
            {{emailAddress}} <i class="fa fa-chevron-down"></i>
            <div class="hoverdrop">
              <button class="btn btn-primary" (click)="logout()">Log Out</button>
            </div>
          </div>
        </li>

      </ul>
    </div>

    <div ui-view></div>
`

const filesMap = {
  componentWithInline: `
  import { Component, ViewChild, ViewContainerRef, OnInit } from '@angular/core';
import { DialogService } from './global/dialog.service';
import { StateService } from '@uirouter/core';
import { AuthService } from './global/auth.service';
import { AppConfigService } from './global/app-config.service';

    @Component({
  selector: 'app-root',
  template: \`${template}\`
  ,
  styles: []
})
export class AppComponent implements OnInit {
  @ViewChild('dialogdiv', { read: ViewContainerRef, static: true }) dialogdiv;

  // data
  emailAddress;
  isAuthenticated;

  constructor(appConfig: AppConfigService,
              public authService: AuthService,
              public $state: StateService,
              private dialog: DialogService
  ) {
    this.emailAddress = appConfig.emailAddress;
    this.isAuthenticated = authService.isAuthenticated();
  }

  ngOnInit() {
    this.dialog.vcRef = this.dialogdiv;
  }

  show() {
    this.dialog.confirm('foo');
  }

  logout() {
    const { authService, $state } = this;
    authService.logout();
    // Reload states after authentication change
    return $state.go('welcome', {}, { reload: true });
  }
}
  `,
  componentWithTemplateUrl: `
    import { Component, ViewChild, ViewContainerRef, OnInit } from '@angular/core';
import { DialogService } from './global/dialog.service';
import { StateService } from '@uirouter/core';
import { AuthService } from './global/auth.service';
import { AppConfigService } from './global/app-config.service';
    @Component({
  selector: 'app-root',
  templateUrl: './my-file.html'
  ,
  styles: []
})
export class AppComponent implements OnInit {
  @ViewChild('dialogdiv', { read: ViewContainerRef, static: true }) dialogdiv;

  // data
  emailAddress;
  isAuthenticated;

  constructor(appConfig: AppConfigService,
              public authService: AuthService,
              public $state: StateService,
              private dialog: DialogService
  ) {
    this.emailAddress = appConfig.emailAddress;
    this.isAuthenticated = authService.isAuthenticated();
  }

  ngOnInit() {
    this.dialog.vcRef = this.dialogdiv;
  }

  show() {
    this.dialog.confirm('foo');
  }

  logout() {
    const { authService, $state } = this;
    authService.logout();
    // Reload states after authentication change
    return $state.go('welcome', {}, { reload: true });
  }
}
  `
} as const;

describe('migrateToAngularRouter', () => {
  let tree: Tree;


  function setup(file: keyof typeof filesMap) {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'my-angular-app', {
      root: `my-angular-app`,
      sourceRoot: `my-angular-app/src`,
      targets: {
      },
    });

    tree.write('package.json', `{"dependencies": {"@uirouter/angular": "13.0.0"}}`);
    tree.write(`my-angular-app/my-file.ts`, filesMap[file]);

    if (file === 'componentWithTemplateUrl') {
      tree.write(`my-angular-app/my-file.html`, template);
      return () => {
        return [
          tree.read('my-angular-app/my-file.ts', 'utf8'),
          filesMap[file],
          tree.read('my-angular-app/my-file.html', 'utf8'),
          template,
        ];
      };
    }

    return () => {
      return [tree.read('my-angular-app/my-file.ts', 'utf8'), filesMap[file]];
    };
  }

  it('should show convert with templateURL', async () => {
    const readContent = setup('componentWithTemplateUrl');
    await migrateToAngularRouter(tree, {});
    const [updated, updatedHtml] = readContent();
    expect(updated).toMatchSnapshot();
    expect(updatedHtml).toMatchSnapshot();
  });

  it('should convert with inline html', async () => {
    const readContent = setup('componentWithInline');
    await migrateToAngularRouter(tree, {});
    const [updated] = readContent();
    expect(updated).toMatchSnapshot();
  });

});