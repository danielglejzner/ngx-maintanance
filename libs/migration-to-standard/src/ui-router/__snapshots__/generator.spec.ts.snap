// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migrateToAngularRouter should convert with inline html 1`] = `
"import { Component, ViewChild, ViewContainerRef, OnInit } from "@angular/core";
import { DialogService } from "./global/dialog.service";
import { StateService } from "@uirouter/core";
import { AuthService } from "./global/auth.service";
import { AppConfigService } from "./global/app-config.service";
@Component({
    selector: "app-root",
    template: "\\n  <div #dialogdiv></div>\\n    <div class=navheader>\\n      <ul *ngIf=isAuthenticated class=nav nav-tabs>\\n\\n        <li routerLinkActive=active> <a [routerLink]=['mymessages'] role=button> Messages </a> </li>\\n        <li routerLinkActive=active> <a [routerLink]=['contacts'] role=button> Contacts </a> </li>\\n        <li routerLinkActive=active> <a [routerLink]=['prefs'] role=button> Preferences </a> </li>\\n\\n        <li class=navbar-right>\\n          <button class=btn btn-primary fa fa-home [routerLink]=['home']></button>\\n          <button style=margin-right: 15px; class=btn btn-primary [routerLink]=['mymessages/compose']>\\n            <i class=fa fa-envelope></i> New Message\\n          </button>\\n        </li>\\n\\n        <li class=navbar-text navbar-right logged-in-user style=margin: 0.5em 1.5em;>\\n          <div>\\n            {{emailAddress}} <i class=fa fa-chevron-down></i>\\n            <div class=hoverdrop>\\n              <button class=btn btn-primary (click)=logout()>Log Out</button>\\n            </div>\\n          </div>\\n        </li>\\n\\n      </ul>\\n    </div>\\n\\n    <router-outlet ></router-outlet>\\n",
    styles: []
})
export class AppComponent implements OnInit {
    @ViewChild("dialogdiv", { read: ViewContainerRef, static: true })
    dialogdiv;
    // data
    emailAddress;
    isAuthenticated;
    constructor(appConfig: AppConfigService, public authService: AuthService, public $state: StateService, private dialog: DialogService) {
        this.emailAddress = appConfig.emailAddress;
        this.isAuthenticated = authService.isAuthenticated();
    }
    ngOnInit() {
        this.dialog.vcRef = this.dialogdiv;
    }
    show() {
        this.dialog.confirm("foo");
    }
    logout() {
        const { authService, $state } = this;
        authService.logout();
        // Reload states after authentication change
        return $state.go("welcome", {}, { reload: true });
    }
}
"
`;

exports[`migrateToAngularRouter should show convert with templateURL 1`] = `
"import { Component, ViewChild, ViewContainerRef, OnInit } from "@angular/core";
import { DialogService } from "./global/dialog.service";
import { StateService } from "@uirouter/core";
import { AuthService } from "./global/auth.service";
import { AppConfigService } from "./global/app-config.service";
@Component({
    selector: "app-root",
    templateUrl: "./my-file.html",
    styles: []
})
export class AppComponent implements OnInit {
    @ViewChild("dialogdiv", { read: ViewContainerRef, static: true })
    dialogdiv;
    // data
    emailAddress;
    isAuthenticated;
    constructor(appConfig: AppConfigService, public authService: AuthService, public $state: StateService, private dialog: DialogService) {
        this.emailAddress = appConfig.emailAddress;
        this.isAuthenticated = authService.isAuthenticated();
    }
    ngOnInit() {
        this.dialog.vcRef = this.dialogdiv;
    }
    show() {
        this.dialog.confirm("foo");
    }
    logout() {
        const { authService, $state } = this;
        authService.logout();
        // Reload states after authentication change
        return $state.go("welcome", {}, { reload: true });
    }
}
"
`;

exports[`migrateToAngularRouter should show convert with templateURL 2`] = `
"
    import { Component, ViewChild, ViewContainerRef, OnInit } from '@angular/core';
import { DialogService } from './global/dialog.service';
import { StateService } from '@uirouter/core';
import { AuthService } from './global/auth.service';
import { AppConfigService } from './global/app-config.service';
    @Component({
  selector: 'app-root',
  templateUrl: './my-file.html'
  ,
  styles: []
})
export class AppComponent implements OnInit {
  @ViewChild('dialogdiv', { read: ViewContainerRef, static: true }) dialogdiv;

  // data
  emailAddress;
  isAuthenticated;

  constructor(appConfig: AppConfigService,
              public authService: AuthService,
              public $state: StateService,
              private dialog: DialogService
  ) {
    this.emailAddress = appConfig.emailAddress;
    this.isAuthenticated = authService.isAuthenticated();
  }

  ngOnInit() {
    this.dialog.vcRef = this.dialogdiv;
  }

  show() {
    this.dialog.confirm('foo');
  }

  logout() {
    const { authService, $state } = this;
    authService.logout();
    // Reload states after authentication change
    return $state.go('welcome', {}, { reload: true });
  }
}
  "
`;
